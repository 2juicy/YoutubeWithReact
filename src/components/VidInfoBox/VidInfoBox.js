import React, { useState } from "react";
import "./VidInfoBox.scss";
import { Button, Image } from "semantic-ui-react";

export default function VidInfoBox() {
  const [toggleCollapse, setToggle] = useState(true);
  let collapse = toggleCollapse ? "collapsed" : "expanded";
  let buttonTitle = toggleCollapse ? "Show More" : "Show Less";

  return (
    <div className="vid-info-box">
      <Image
        className="channel-img"
        src="http://via.placeholder.com/48x48"
        circular
      />
      <div className="vid-info">
        <div className="channel-name">Channel Name</div>
        <div className="vid-publish-date">Fri 13, 2019</div>
      </div>
      <Button color="youtube">12.3K Subscribe</Button>
      <div className="vid-description">
        <div className={collapse}>
          <p>Paragraph...</p>
          <p>Paragraph...</p>
          <p>Paragraph...</p>
          <p>Paragraph...</p>
          <p>Paragraph...</p>
        </div>
        <Button onClick={() => setToggle(state => !state)} compact>
          {buttonTitle}
        </Button>
      </div>
    </div>
  );
}
