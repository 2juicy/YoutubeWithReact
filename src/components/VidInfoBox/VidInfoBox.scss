.vid-info-box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  row-gap: 16px;
}
