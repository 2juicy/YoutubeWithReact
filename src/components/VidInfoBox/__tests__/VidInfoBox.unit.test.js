import React from "react";
import { shallow } from "enzyme";
import VidInfoBox from "../VidInfoBox";

describe("VidInfoBox", () => {
  test("renders", () => {
    const wrapper = shallow(<VidInfoBox />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders expanded", () => {
    const wrapper = shallow(<VidInfoBox hook={() => setToggle(false)} />);
    expect(wrapper).toMatchSnapshot();
  });
});
