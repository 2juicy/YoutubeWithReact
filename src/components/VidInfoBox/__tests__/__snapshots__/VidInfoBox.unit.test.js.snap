// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VidInfoBox renders 1`] = `
<div
  className="vid-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-img"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="vid-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="vid-publish-date"
    >
      Fri 13, 2019
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    12.3K Subscribe
  </Button>
  <div
    className="vid-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`VidInfoBox renders expanded 1`] = `
<div
  className="vid-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-img"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="vid-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="vid-publish-date"
    >
      Fri 13, 2019
    </div>
  </div>
  <Button
    as="button"
    color="youtube"
  >
    12.3K Subscribe
  </Button>
  <div
    className="vid-description"
  >
    <div
      className="collapsed"
    >
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
      <p>
        Paragraph...
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;
